src/clarity-angular/datagrid/render/column-resizer.ts

```diff
@@ -0,0 +1,190 @@
+    }
+
+    ngOnDestroy() {
+        this.columnResizeEvents.map(event => event());
+    }
+
+
+    addDragListener(): Function[] {
+
+        this.draggerEl = this.columnEl.querySelector("button.separator-dragger");
+        this.draggerTracker = this.columnEl.querySelector(".separator-tracker");
+
+        this.draggerElWidth = this.domAdapter.clientRectWidth(this.draggerEl);
+
+        this.columnMinWidth = this.domAdapter.minWidth(this.columnEl);
+
+        return this.listenDrag();
+    }
+
+    listenDrag(): Function[] {
+
+        let mouseDownListener: any;
+        let mouseMoveListener: any;
+        let mouseUpListener: any;
+
+        let touchStartListener: any;
+        let touchMoveListener: any;
+        let touchEndListener: any;
+
+        mouseDownListener = this.renderer.listen(this.draggerEl, "mousedown", () => {
+            this.onDragStart();
+
+            mouseMoveListener = this.renderer.listenGlobal("document", "mousemove", (mouseMoveEvent: any) => {
+                this.onDragMove(mouseMoveEvent);
+            });
+
+            mouseUpListener = this.renderer.listenGlobal("document", "mouseup", () => {
+
+                //Unsubscribing from mouseMoveListener
+                mouseMoveListener();
+                this.onDragEnd();
+
+                //Unsubscribing from itself
+                mouseUpListener();
+            });
+        });
+
+        touchStartListener = this.renderer.listen(this.draggerEl, "touchstart", () => {
+            this.onDragStart();
+
+            touchMoveListener = this.renderer.listenGlobal("document", "touchmove", (touchMoveEvent: any) => {
+                this.onDragMove(touchMoveEvent);
+            });
+
+            touchEndListener = this.renderer.listenGlobal("document", "touchend", () => {
+
+                //Unsubscribing from mouseMoveListener
+                touchMoveListener();
+                this.onDragEnd();
+
+                //Unsubscribing from itself
+                touchEndListener();
+            });
+        });
+
+        //returning mouseDownListener and touchStartListener so they can be unsubscribed.
+        return [mouseDownListener, touchStartListener];
+    }
+
+    onDragStart(): void {
+
+        this.renderer.setElementStyle(this.draggerTracker, "display", "block");
+
+        this.dragDistancePositionX = 0;
+        this.columnRectWidth = this.domAdapter.clientRectWidth(this.columnEl);
+        this.pageStartPositionX = this.domAdapter.clientRectRight(this.columnEl);
+
+    }
+
+    onDragMove(moveEvent: any): void {
+        let pageMovePosition = moveEvent.pageX || moveEvent.changedTouches[0].pageX;
+
+        this.dragDistancePositionX = this.getPositionWithinMax(pageMovePosition - this.pageStartPositionX);
+
+        this.renderer.setElementStyle(this.draggerTracker, "right", -1 * this.dragDistancePositionX + "px");
+        this.renderer.setElementStyle(document.body, "cursor", "col-resize");
```

Comment:
Shouldn't you set the cursor on the whole body in `onDragStart()`? It seems counterproductive to do many times per second when you only need to do it once.